### PR TITLE
rpm: 4.18.1 -> 4.20.0

### PR DESCRIPTION
**Pull request by baloo**
Depends #354645

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by emilazy** [comment]
This will re‐break the build on Darwin, sadly, even as it incorporates upstream patches for the platform. IIRC @reckenrode said that the new Meson build system needed fixes when he looked at doing the update, though it is hard to tell as there are other blockers here preventing testing that (unconditional `bubblewrap` and `systemd` inputs, e.g.).

**Comment by reckenrode** [comment]
Upstream no longer does feature detection. Features that are not supported on a platform (or not wanted) have to be disabled manually instead of relying on the library/function not being available. Of course, the unconditional systemd stuff is also a problem for non-Linux.

**Comment by baloo** [comment]
I got it to build on aarch64-darwin and x86-64-darwin. I did not try to run the binaries though.

File: pkgs/tools/package-management/rpm/default.nix [hunk 1]
```diff
@@ -1,113 +1,86 @@
...
-, python
+, python3
 , lua
 , llvmPackages
 , sqlite
 , zstd
 , libcap
 , apple-sdk_13
 , darwinMinVersionHook
-, gnupg
```
Comment (baloo) [inline pkgs/tools/package-management/rpm/default.nix:29]: gnupg is required for `rpmsign`

**Comment by baloo** [comment]
I missed the sync from master to staging by a couple hours, I will rebase tomorrow

**Comment by baloo** [comment]
rpm-sequoia needs to be fixed on macos:
```
dyld[69225]: Library not loaded: /nix/store/cz9sppfrs9ag34lq2pq6wmhspwcnvbxq-rpm-sequoia-1.7.0/lib/librpm_sequoia.1.7.0.dylib
  Referenced from: <0EA94AC5-F54A-3791-9374-4092232F33A1> /nix/store/mak6qp2pvgz19dn0fb0pr2lksgca43qp-rpm-4.20.0/lib/librpmio.10.2.0.dylib
  Reason: tried: '/nix/store/cz9sppfrs9ag34lq2pq6wmhspwcnvbxq-rpm-sequoia-1.7.0/lib/librpm_sequoia.1.7.0.dylib' (no such file), '/System/Vol>
cpio: premature end of archive
```

**Comment by msanft** [comment]
I think [these commits](https://github.com/rpm-software-management/rpm/commits/4d621e48e07ccf7aac68b4ceabf14b2fd827b2a4/plugins/unshare.c) break execution of `rpm` within the Nix sandbox. Did you run into the same thing by any chance @baloo?

**Comment by emilazy** [comment]
Ideally we'd disable the plugin at runtime instead.

**Comment by msanft** [comment]
> Ideally we'd disable the plugin at runtime instead.

That'd be the best way certainly, but not all users of `rpm` (e.g. `tdnf`) expose such a knob. See some more detailed reasoning in #371387

**Comment by takeda** [comment]
@baloo Is it possible that the latest rpmbuild version is causing this? https://github.com/NixOS/nixpkgs/issues/369167

**Pull request merged**